### PR TITLE
clustermesh: Improve troubleshooting ability

### PR DESCRIPTION
**Pull request by tgraf**
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/9081)

### CODE REVIEW

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
test-docs-please

File: Documentation/gettingstarted/clustermesh.rst [hunk 1]
```diff
@@ -359,54 +359,172 @@ the ``--cluster-name`` agent option or ``cluster-name`` ConfigMap option.
...
- * Look for the following message to ensure that ClusterMesh is initialized:
+Use the following list of steps to troubleshoot issues with ClusterMesh:
 
-   .. code:: bash
+Generic
+=======
+
+ #. Validate that the ``cilium-xxx`` as well as the ``cilium-operator-xxx` pods
+    are healthy and ready. Is is important that the ``cilium-operator`` is
+    healthy as well as it is responsible to synchronize state from the local
```
Comment (ianvernon) [inline Documentation/gettingstarted/clustermesh.rst]: responsible for synchronizing

File: Documentation/gettingstarted/clustermesh.rst [hunk 2]
```diff
@@ -359,54 +359,172 @@ the ``--cluster-name`` agent option or ``cluster-name`` ConfigMap option.
...
-   it will list the Kubernetes services and endpoints of the local cluster.  It
-   will also have a section ``externalEndpoints`` which must list all endpoints
-   of remote clusters.
+    * Are services available in the kvstore of each cluster? You can confirm
+      this by running ``cilium kvstore get --recursive
+      cilium/state/services/v1/``.
+ 
+    * Run ``cilium debuginfo`` and look for the section "k8s-service-cache". In
+      that section, you will find the contents of the service correlation
+      cache.  it will list the Kubernetes services and endpoints of the local
```
Comment (ianvernon) [inline Documentation/gettingstarted/clustermesh.rst:527]: Provide an example of the output here.

**Review by ianvernon — CHANGES_REQUESTED** [review]
Small nits. Great work!

File: Documentation/gettingstarted/clustermesh.rst [hunk 3]
```diff
@@ -359,54 +359,172 @@ the ``--cluster-name`` agent option or ``cluster-name`` ConfigMap option.
...
- * Check the Cilium pod logs for ``level=error`` and ``level=warning`` messages.
- * Look for the following message to ensure that ClusterMesh is initialized:
+Use the following list of steps to troubleshoot issues with ClusterMesh:
 
-   .. code:: bash
+Generic
+=======
+
+ #. Validate that the ``cilium-xxx`` as well as the ``cilium-operator-xxx` pods
+    are healthy and ready. Is is important that the ``cilium-operator`` is
```
Comment (joestringer) [inline Documentation/gettingstarted/clustermesh.rst]:
```suggestion
    are healthy and ready. It is important that the ``cilium-operator`` is
```

File: Documentation/gettingstarted/clustermesh.rst [hunk 4]
```diff
@@ -359,54 +359,172 @@ the ``--cluster-name`` agent option or ``cluster-name`` ConfigMap option.
...
+    healthy as well as it is responsible to synchronize state from the local
+    cluster into the kvstore. If this fails, check the logs of these pods to
+    track the reason for failure.
+
+ #. Validate that the ClusterMesh subsystem is initialized by looking for a log
```
Comment (joestringer) [inline Documentation/gettingstarted/clustermesh.rst]:
```suggestion
 #. Validate that the ClusterMesh subsystem is initialized by looking for a ``cilium-agent`` log
```

File: Documentation/gettingstarted/clustermesh.rst [hunk 5]
```diff
@@ -359,54 +359,172 @@ the ``--cluster-name`` agent option or ``cluster-name`` ConfigMap option.
...
-   the contents of the directory ``/var/lib/cilium/clustermesh/``. It must
-   contain a configuration file for each remote cluster along with all the
-   required SSL certificates and keys. The filenames must match the cluster
-   names as provided by the ``--cluster-name`` argument or ``cluster-name``
-   ConfigMap option. If the directory is empty or incomplete, regenerate the
-   secret again and ensure that the secret is correctly mounted into the
-   DaemonSet.
+ #. Validate that the configuration for remote clusters is picked up correctly.
+    For each remote cluster, an info log message ``New remote cluster
+    configuration`` along with the remote cluster name must be logged.
```
Comment (joestringer) [inline Documentation/gettingstarted/clustermesh.rst]:
```suggestion
    configuration`` along with the remote cluster name must be logged in the ``cilium-agent`` logs.
```

File: Documentation/gettingstarted/clustermesh.rst [hunk 6]
```diff
@@ -359,54 +359,172 @@ the ``--cluster-name`` agent option or ``cluster-name`` ConfigMap option.
...
+      ConfigMap option. If the directory is empty or incomplete, regenerate the
+      secret again and ensure that the secret is correctly mounted into the
+      DaemonSet.
 
- * When using global services, ensure that the ``cilium-operator`` deployment
-   is running and healthy. It is responsible to propagate Kubernetes services
-   into the kvstore. You can validate the correct functionality of the operator
-   by running:
+ #. Validate that the connection to the remote cluster could be established.
+    You will see a log message like this for each remote cluster:
```
Comment (joestringer) [inline Documentation/gettingstarted/clustermesh.rst]:
```suggestion
    You will see a log message like this in ``cilium-agent`` logs for each remote cluster:
```

File: Documentation/gettingstarted/clustermesh.rst [hunk 7]
```diff
@@ -359,54 +359,172 @@ the ``--cluster-name`` agent option or ``cluster-name`` ConfigMap option.
...
+    * In each cluster, check that the kvstore contains information about
+      *local* nodes by running:
+
+      .. code:: bash
+
+          cilium kvstore get --recursive cilium/state/nodes/v1/
+
+      .. note::
+
+	 The kvstore will only contain nodes of the **local cluster**. It will
```
Comment (joestringer) [inline Documentation/gettingstarted/clustermesh.rst]: nit: The tabbing seems a bit weird to me in github interface. As long as the indentation looks right then that's fine.

**Review by joestringer — APPROVED** [review]
Some minor nits, but this is really helpful.

**Comment by tgraf** [comment]
test-docs-please

**Pull request merged**